rules_gapic/gapic.bzl

```diff
@@ -12,37 +12,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+def _set_args(arg, arg_name, args, inputs = None):
+    if not arg:
+        return
+    args.append("%s%s" % (arg_name, arg.files.to_list()[0].path if hasattr(arg, "files") else arg))
+    if inputs != None:
+        inputs.append(arg.files.to_list()[0])
+
 def _gapic_srcjar_impl(ctx):
-    output = ctx.outputs.output
-    optional_arguments = []
-    optional_action_inputs = []
-
-    if ctx.file.service_yaml:
-        optional_arguments.append("--service_yaml=%s" % ctx.file.service_yaml.path)
-        optional_action_inputs.append(ctx.file.service_yaml)
-    if ctx.attr.package_yaml2:
-        optional_arguments.append("--package_yaml2=%s" % ctx.file.package_yaml2.path)
-        optional_action_inputs.append(ctx.file.package_yaml2)
-
-    if ctx.attr.artifact_type.find("DISCOGAPIC") >= 0:
-        optional_arguments.append("--discovery_doc=%s" % ctx.file.src.path)
+    arguments = []
+    inputs = []
+
+    attr = ctx.attr
+    if attr.artifact_type:
```

Comment:
This branch requires the `gapic_yaml` and `language`. Add a `fail` call to ensure its presence.

Reply:
Added for language. For `gapic_yaml` it is to support cases when we don't need any gapic_yaml (even v2).